### PR TITLE
dbeaver/pro#3137 Clear embedded browser cookies

### PR DESCRIPTION
**Pull request by ShadelessFox**
image

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

**Review by ivgag — APPROVED** [review]

**Comment by MashaKorax** [comment]
an error after applaying reset cookies
```
org.jkiss.dbeaver.DBException: Class 'org.jkiss.dbeaver.ui.browser.handlers.BrowserClearCookiesHandler' not found
	at org.jkiss.dbeaver.model.impl.AbstractDescriptor$ObjectType.checkObjectClass(AbstractDescriptor.java:231)
	at org.jkiss.dbeaver.ui.app.standalone.actions.ClearHistoryHandler$HandlerDescriptor.getHandler(ClearHistoryHandler.java:165)
	at org.jkiss.dbeaver.ui.app.standalone.actions.ClearHistoryHandler.execute(ClearHistoryHandler.java:79)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:277)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:299)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:233)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:174)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:165)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:204)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:444)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:442)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:464)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:292)
	at com.dbeaver.application.DBeaverApplicationAdvanced.start(DBeaverApplicationAdvanced.java:97)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
```
https://github.com/user-attachments/assets/f9d650a4-9856-48d0-a4e7-6e629f90116a

**Comment by MashaKorax** [comment]
```
java.lang.IllegalArgumentException: Argument not valid [WebView2: environment not initialized for current display]
	at org.eclipse.swt.SWT.error(SWT.java:4903)
	at org.eclipse.swt.browser.Edge.getCookieManager(Edge.java:283)
	at org.eclipse.swt.browser.Edge.lambda$0(Edge.java:69)
	at org.eclipse.swt.browser.WebBrowser.clearSessions(WebBrowser.java:265)
	at org.eclipse.swt.browser.Browser.clearSessions(Browser.java:220)
	at org.jkiss.dbeaver.ui.browser.handlers.BrowserClearCookiesHandler.run(BrowserClearCookiesHandler.java:26)
	at org.jkiss.dbeaver.ui.UIUtils.lambda$20(UIUtils.java:1970)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:124)
```

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.swt.linux/src/org/eclipse/swt/browser/DBeaverBrowserHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    private DBeaverBrowserHelper() {
+    }
+
+    public static void clearCookies(Browser browser) {
+        Objects.requireNonNull(browser, "browser cannot be null");
+        if (!WebKitGTK.LibraryLoaded) {
+            return;
+        }
+        if (WebKitGTK.webkit_get_minor_version() < 16) {
+            System.err.println("SWT WebKit: clear sessions only supported on WebKitGtk version 2.16 and above.");
```
Comment (ivgag) [inline plugins/org.jkiss.dbeaver.ui.swt.linux/src/org/eclipse/swt/browser/DBeaverBrowserHelper.java]: Did you intentionally use the standard error output here?

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.swt.linux/src/org/eclipse/swt/browser/DBeaverBrowserHelper.java]: It was a copy-paste from the original source. I don't think it's even needed but I'll fix it regardless.

**Comment by uslss** [comment]
windows - verified

**Comment by Matvey16** [comment]
MacOS - verified

**Pull request merged**